platform/src/main/java/org/hillview/storage/CassandraSSTableLoader.java

```diff
@@ -345,42 +295,42 @@ public ITable nonLazyLoading() {
                     i = 0;
                     /* Extracting the value for each column from a single row  */
                     for (ColumnData cd : row) {
-                        if (cd.column().isSimple()) {
-                            cell = (Cell) cd;
-                            cellType = cell.column().cellValueType();
-                            value = cellType.getSerializer().deserialize(cell.value());
-                        } else {
-                            throw new RuntimeException("[Hillview can't convert complex data]");
+                        /* Filter the column to only load the one marked true by columnToLoad[] */
+                        if (columnToLoad[i]){
+                            if (cd.column().isSimple()){
+                                cell = (Cell) cd;
+                                cellType = cell.column().cellValueType();
+                                value = cellType.getSerializer().deserialize(cell.value());
+                            }else{
+                                /* Hillview won't process the complex data */
+                                throw new RuntimeException("[Hillview can't convert complex data]");
+                            }
+                            line[i] = value; 
                         }
-                        line[i] = value; 
                         i++;
                     }
-
                     /* Append each line to Hillview column */
                     for (this.currentColumn = 0; this.currentColumn < line.length; this.currentColumn++) {
-                        this.columns[this.currentColumn].append(line[this.currentColumn]);
+                        columns[this.currentColumn].append(line[this.currentColumn]);
                     }
-                    this.currentRow++;
-
-                }else if (unfiltered instanceof RangeTombstoneMarker){
+                } else if (unfiltered instanceof RangeTombstoneMarker) {
```

Comment:
What are these? Is there a reason to process them? Do they indicate a deleted row?

Reply:
Yes, it's a marker that the item is deleted